grim_test.go

```diff
@@ -8,11 +8,50 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+	"bytes"
+	"strings"
 )
 
 // Copyright 2015 MediaMath <http://www.mediamath.com>.  All rights reserved.
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
+
+func TestTruncatedGrimServerID(t *testing.T) {
+	var buf bytes.Buffer
+	logger := log.New(&buf, "", log.Lshortfile)
+
+	tempDir, _ := ioutil.TempDir("", "TestTruncatedGrimServerID")
+	defer os.RemoveAll(tempDir)
+
+	empty := "empty"
+	hugeString := "12345678901234567890"
+
+	grimConfig := &config{
+		ResultRoot:    &empty,
+		WorkspaceRoot: &empty,
+		AWSRegion:     &empty,
+		AWSKey:        &empty,
+		AWSSecret:     &empty,
+		GrimServerID:  &hugeString,
+	}
+
+	configJs, _ := json.Marshal(grimConfig)
+	ioutil.WriteFile(filepath.Join(tempDir, "config.json"), configJs, 0644)
```

Comment:
Unless I'm missing something this test would get a lot simpler if we just store the generated `config.json` in the repo.